prow/flagutil/github.go

```diff
@@ -102,7 +105,7 @@ func (o *GitHubOptions) GitHubClient(secretAgent *secret.Agent, dryRun bool) (cl
 
 // GitClient returns a Git client.
 func (o *GitHubOptions) GitClient(secretAgent *secret.Agent, dryRun bool) (client *git.Client, err error) {
-	client, err = git.NewClient()
+	client, err = git.NewClient(o.gitEndpoint)
```

Comment:
This won't work, especially if you are passing endpoints like `api.github.com` and `github.mycorp/api/v3`

Reply:
Sorry, mis-read, I thought you were re-using the github endpoint